owncloudData/src/test/java/com/owncloud/android/data/storage/ScopedStorageProviderTest.kt

```diff
@@ -0,0 +1,304 @@
+        val expectedPath = absolutePath + File.separator + rootFolderName
+        val actualPath = scopedStorageProvider.getRootFolderPath()
+        assertEquals(expectedPath, actualPath)
+
+        verify(exactly = 1) {
+            scopedStorageProvider.getPrimaryStorageDirectory()
+            filesDir.absolutePath
+        }
+
+    }
+
+    @Test
+    fun `get default save path if there are space - ok - should return the String with spaces`() {
+        mockkStatic(Uri::class)
+        every { Uri.encode(accountName, "@") } returns uriEncoded
+
+        val accountDirectoryPath = absolutePath + File.separator + rootFolderName + File.separator + uriEncoded
+
+        val expectedPath = accountDirectoryPath + File.separator + spaceId + File.separator + remotePath
+        val actualPath = scopedStorageProvider.getDefaultSavePathFor(accountName, remotePath, spaceId)
+
+        assertEquals(expectedPath, actualPath)
+
+        verify(exactly = 1) {
+            scopedStorageProvider.getPrimaryStorageDirectory()
+            filesDir.absolutePath
+        }
+    }
+
+    @Test
+    fun `get default save path if there are not space - ok - should return the String without spaces`() {
+        val spaceId = null
+
+        mockkStatic(Uri::class)
+        every { Uri.encode(accountName, "@") } returns uriEncoded
+
+        val accountDirectoryPath = absolutePath + File.separator + rootFolderName + File.separator + uriEncoded
+
+        val expectedPath = accountDirectoryPath + remotePath
+        val actualPath = scopedStorageProvider.getDefaultSavePathFor(accountName, remotePath, spaceId)
+
+        assertEquals(expectedPath, actualPath)
+
+        verify(exactly = 1) {
+            scopedStorageProvider.getPrimaryStorageDirectory()
+            filesDir.absolutePath
+        }
+    }
+
+    @Test
+    fun `get expected remote path with parent variable with separator - ok - should return String expected remote path`() {
+
+        val isFolder = true
+        val parent = "\\storage\\emulated\\0\\owncloud\\"
+
+        val expectedPath = conditionsExpectedRemotePath(parent, newName, isFolder)
+        val actualPath = scopedStorageProvider.getExpectedRemotePath(remotePath, newName, isFolder)
+
+        assertEquals(expectedPath, actualPath)
+    }
+
+    @Test
+    fun `get expected remote path with parent variable without separator - ok - should return String expected remote path`() {
+
+        val isFolder = true
+        val parent = "\\storage\\emulated\\0\\owncloud"
+
+        val expectedPath = conditionsExpectedRemotePath(parent, newName, isFolder)
+        val actualPath = scopedStorageProvider.getExpectedRemotePath(remotePath, newName, isFolder)
+
+        assertEquals(expectedPath, actualPath)
+    }
+
+    @Test
+    fun `get expected remote path with parent variable without separator and it is not a folder - ok - should return String expected remote path`() {
+        val newName = "onwcloudNewName.txt"
```

Comment:
I think so because it is another path in the method. If there are folders, the code will do a condition, if not it will return something. In any case, it does not hurt to have this test on.